src/Type/Php/RegexArrayShapeMatcher.php

```diff
@@ -724,38 +726,33 @@ private function getPatternType(Expr $patternExpr, Scope $scope): Type
 	 */
 	private function resolvePatternConcat(Expr\BinaryOp\Concat $concat, Scope $scope): Type
 	{
-		if (
-			$concat->left instanceof Expr\FuncCall
-			&& $concat->left->name instanceof Name
-			&& $concat->left->name->toLowerString() === 'preg_quote'
-		) {
-			$left = new ConstantStringType('');
-		} elseif ($concat->left instanceof Expr\BinaryOp\Concat) {
-			$left = $this->resolvePatternConcat($concat->left, $scope);
-		} else {
-			$left = $scope->getType($concat->left);
-		}
-
-		if (
-			$concat->right instanceof Expr\FuncCall
-			&& $concat->right->name instanceof Name
-			&& $concat->right->name->toLowerString() === 'preg_quote'
-		) {
-			$right = new ConstantStringType('');
-		} elseif ($concat->right instanceof Expr\BinaryOp\Concat) {
-			$right = $this->resolvePatternConcat($concat->right, $scope);
-		} else {
-			$right = $scope->getType($concat->right);
-		}
+		$resolveConcat = static function (Expr $expr) use (&$resolveConcat, $scope) {
```

Comment:
These are usually memory leaks. It'd be nicer to have this as a private method.

Reply:
I want with a inline class, hopefully that works for you.
didn't have a better idea on how to get the lexical $scope variable into the callable